File: ControllerAnimations/ControllerAnimations/ViewController.m
//
//  ViewController.m
//  ControllerAnimations
//
//  Created by João Gabriel Soares Esteves on 03/11/14.
//  Copyright (c) 2014 JoaoTech. All rights reserved.
//

#import "ViewController.h"
#import "OpcoesController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)showDissolve:(id)sender {
    [self mostraControllerComAnimacao:UIModalTransitionStyleCrossDissolve];
}

- (IBAction)showPageCurl:(id)sender {
    [self mostraControllerComAnimacao:UIModalTransitionStylePartialCurl];
}

- (IBAction)showVertical:(id)sender {
    [self mostraControllerComAnimacao:UIModalTransitionStyleCoverVertical];
}

- (IBAction)showHorizontal:(id)sender {
    [self mostraControllerComAnimacao:UIModalTransitionStyleFlipHorizontal];
}

- (void) mostraControllerComAnimacao:(UIModalTransitionStyle) estilo {
    OpcoesController *c = [[OpcoesController alloc] init];
    c.modalTransitionStyle = estilo;
    
    //    [self presentModalViewController:c animated:YES completion:nil];
    [self presentViewController:c animated:YES completion:nil];
}
@end

File: HelloSwift/HelloSwift/ViewController.swift
//
//  ViewController.swift
//  HelloSwift
//
//  Created by João Gabriel Soares Esteves on 19/10/14.
//  Copyright (c) 2014 JoaoTech. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var workingView: UIView!
    
    @IBAction func hideView(sender: AnyObject) {
        workingView.alpha = 0;
    }
    
    @IBAction func showView(sender: AnyObject) {
        workingView.alpha = 1;
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

